AudioRecorder.py

```diff
@@ -34,17 +39,20 @@ def __init__(self):
 
 class DefaultSpeakerRecorder(BaseRecorder):
     def __init__(self):
-        with pyaudio.PyAudio() as p:
-            wasapi_info = p.get_host_api_info_by_type(pyaudio.paWASAPI)
-            default_speakers = p.get_device_info_by_index(wasapi_info["defaultOutputDevice"])
-            
-            if not default_speakers["isLoopbackDevice"]:
-                for loopback in p.get_loopback_device_info_generator():
-                    if default_speakers["name"] in loopback["name"]:
-                        default_speakers = loopback
-                        break
-                else:
-                    print("[ERROR] No loopback device found.")
+        if os.name == 'nt':
+            with pyaudio.PyAudio() as p:
+                wasapi_info = p.get_host_api_info_by_type(pyaudio.paWASAPI)
+                default_speakers = p.get_device_info_by_index(wasapi_info["defaultOutputDevice"])
+                if not default_speakers["isLoopbackDevice"]:
+                    for loopback in p.get_loopback_device_info_generator():
+                        if default_speakers["name"] in loopback["name"]:
+                            default_speakers = loopback
+                            break
+                    else:
+                        print("[ERROR] No loopback device found.")
+        else:
+            p = pyaudio.PyAudio()
+            default_speakers = p.get_device_info_by_index(0)
```

Comment:
This line is the culprit. I had my LG HDR 4K external monitor plugged into my macbook and this causes the issue I described in our conversation. Since we get `p.get_device_info_by_index(0)` and my LG HDR 4K is detected as the first index, this causes the error.

<img width="475" alt="CleanShot_2023-05-30_21 59 09_System Settings_000850@2x" src="https://github.com/SevaSk/ecoute/assets/16494335/750801a8-aa7c-441e-ab6d-916cf7c421bd">

If I unplug the device or [keep it plugged in and] switch the index to 1 to match the appropriate index of my default speakers, the issue is resolved.

Reply:
Perhaps we can update the documentation or add a GUI to select appropriate default speakers?